### ISSUE TITLE
not working on node 10.5

### ISSUE THREAD

**Issue by twawszczak** [tschaub/mock-fs#238]
after fs module refactors and changes

**Comment by demosdemon** [comment]
ahh, this is my problem!

from `<node_internals>/js.js`
```js
function readFileAfterOpen(err, fd) {
  const context = this.context;

  if (err) {
    context.callback(err);
    return;
  }

  context.fd = fd;

  const req = new FSReqWrap();
  req.oncomplete = readFileAfterStat;
  req.context = context;
  binding.fstat(fd, false, req);
}
```

from `mock-fs`
https://github.com/tschaub/mock-fs/blob/5b165be6b7280b7c9ddaab9adac89a6236d123d0/lib/binding.js#L357-L380

**Comment by jimthedev** [comment]
If anyone has any good ideas on how to solve this I'd be willing to put in a PR.

**Comment by tomhughes** [comment]
I've opened #240 which fixes the issues with `stat`/`fstat`/`lstat` but there is a second set of problems with `createReadStream` and `createWriteStream` which I have not been able to find a fix for so far.

Basically the problem is that after restoring the mocked fs and then starting a new one writing to a stream created against the new mock fs will wind up calling back into `writeBuffers` with an fd from the new context but with `this` bound to the old context which leads the fd not being found. I have so far been unable to determine exactly why however...

**Comment by tschaub** [comment]
Stat calls now work (thanks @tomhughes).  I need to do more debugging to figure out how to handle the read after open.

**Comment by mistic** [comment]
@tschaub @tomhughes any news on solving the streams bug in Node v10.5+? This just become urgent for my current use case

**Comment by tschaub** [comment]
This should be addressed on the 4.8.0 release (thanks @huochunpeng, see #260).
